lib/puma/server.rb

```diff
@@ -189,6 +192,7 @@ def run(background=true)
 
       @thread_pool = ThreadPool.new(@min_threads,
                                     @max_threads,
+                                    Proc.new { signal_all_worker_threads_free },
```

Comment:
Why did you choose to pass a Proc here instead of just passing the mutex?

Reply:
Because I needed to pass at least 2 args: the mutex + the boolean itself. And this way the `ThreadPool` doesn't need to know what the `Server` is doing, it just tells "_hey, all my workers are free, do whatever you want_".